icu4c/source/i18n/nfrule.h

```diff
@@ -109,15 +109,15 @@ class NFRule : public UMemory {
                      int32_t startingAt, int32_t* resultCount) const;
 
 private:
-    int64_t baseValue;
-    int32_t radix;
-    int16_t exponent;
-    char16_t decimalPoint;
-    UnicodeString fRuleText;
-    NFSubstitution* sub1;
-    NFSubstitution* sub2;
-    const RuleBasedNumberFormat* formatter;
-    const PluralFormat* rulePatternFormat;
+    int64_t baseValue {  };
+    int32_t radix { 10 };
+    int16_t exponent {  };
+    char16_t decimalPoint {  };
+    UnicodeString ruleText;
+    NFSubstitution* sub1 {  };
+    NFSubstitution* sub2 {  };
+    const RuleBasedNumberFormat* formatter {  };
+    const PluralFormat* rulePatternFormat {  };
```

Comment:
I had to upgrade this to C++11 syntax. It's just too cumbersome to copy and paste the exact same data member initialization in the same order in multiple constructors.